main.c

```diff
@@ -2,7 +2,8 @@
 // Copyright (c) 2019 Michael Steil
 // All rights reserved. License: 2-clause BSD
 
-#define _XOPEN_SOURCE   600
+#define SDL_MAIN_HANDLED
+#define _XOPEN_SOURCE 600
```

Comment:
Why is this required?

Reply:
If you want a pure GUI app in Windows (i.e. with no console), you need to define a `WinMain` function. SDL offers the posibility of using just the same `main` for all platforms, by `#define`ing `main` as `SDL_main` and providing a stub `main` and `WinMain` in `SDLmain.lib`. It happens something similar in macOS, but there you compile `SDLmain.m` (which provides an Objective-C entrypoint) into your project.

Since x16emu does not need to be a pure GUI program and uses the console to provide logs and messages and so, when you said some *nix distros do not provide `SDLmain.a`, I thought it appropiate to just get rid of that dependence. But when you do that, you need to define `SDL_MAIN_HANDLED` before including SDL so the SDL headers do not `#define main`.

Reply:
I think that he asked about the `_XOPEN_SOURCE` macro.

Reply:
Oh. No idea about that.

Reply:
Nevermind my remark.
I did not notice that you modified the number of spaces in that define.